### ISSUE TITLE
Ignore HTML Comment in Markdown before considering YAML Frontmatter header

### ISSUE THREAD

**Issue by vorburger** [markdownlint/markdownlint#515]
## Description

When a Markdown file starts with HTML Comment, e.g. for a LICENSE sort of header, then markdownlint doesn't consider any YAML Frontmatter that follows that license comment header, as I would expect it should (because it's just a comment).

I do fully understand that the exact parsing rules for YAML Frontmatter headers in Markdown are somewhat murky. But at least in a tool that I am writing myself, I am (a) just ignoring any HTML comments at the beginning of the file; (b) then process any Frontmatter in YAML; (c) then consider the rest of the file.

Would you potentially be willing to make markdownlint behave the same way? Or would a Pull Request for this possibly we welcome? Or do you feel extremely very strongly that this is totally invalid MD?

**MDL Version**

markdownlint-cli2 v0.17.1 (markdownlint v0.37.3)

### Expected Behavior

The attached `test-NOK.md` should pass markdownlint just like the attached `test-OK.md` does.

### Actual Behavior

The attached `test-NOK.md` seems to confuse the hell out of markdownlint, just because of that license header in comment, check this out:

```shell
$ markdownlint-cli2 test-OK.md
markdownlint-cli2 v0.17.1 (markdownlint v0.37.3)
Finding: test-OK.md
Linting: 1 file(s)
Summary: 0 error(s)

$ markdownlint-cli2 test-NOK.md
markdownlint-cli2 v0.17.1 (markdownlint v0.37.3)
Finding: test-NOK.md
Linting: 1 file(s)
Summary: 5 error(s)
test-NOK.md:19 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "---"]
test-NOK.md:20 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Above] [Context: ""@context": https://enola.dev/spec.jsonld"]
test-NOK.md:20:13 MD034/no-bare-urls Bare URL used [Context: "https://enola.dev/spec.jsonld"]
test-NOK.md:21:9 MD034/no-bare-urls Bare URL used [Context: "https://www.vorburger.ch"]
test-NOK.md:25 MD003/heading-style Heading style [Expected: setext; Actual: atx]
```

## Replication Case

[test-NOK.md](https://github.com/user-attachments/files/18312395/test-NOK.md)
[test-OK.md](https://github.com/user-attachments/files/18312396/test-OK.md)

**Comment by vorburger** [comment]
FYI: Between having to ditch Markdownlint or not having License headers in a bunch of MD files, I'm temporarily going to opt for the latter instead of the former in a project, but this doesn't seem like a great solution.

**Comment by DavidAnson** [comment]
It looks like you meant to open this issue here instead: https://github.com/DavidAnson/markdownlint-cli2

You can customize the regular expression used for matching front matter via the `frontMatter` configuration property documented there and (in slightly more detail) by the underlying library: https://github.com/DavidAnson/markdownlint?tab=readme-ov-file#optionsfrontmatter
